src/odemis/driver/keysight.py

```diff
@@ -0,0 +1,333 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 29 Apr 2024
+
+@author: Canberk Akin
+
+Copyright © 2024 Canberk Akin, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+import logging
+import re
+import time
+import socket
+
+from odemis import model
+from odemis.model import HwError
+from odemis.util import to_str_escape
+from typing import List, Tuple
+
+
+class TrueForm(model.Emitter):
+    def __init__(self, name, role, address, channels: List[int],
+                 limits: List[Tuple[float, float]], **kwargs):
+        """ Initializes the Keysight 33600 series Arbitrary Waveform Generator.
+        :param name: (str) as in Odemis
+        :param role: (str) as in Odemis
+        :param address: # “fake” (to start the simulator) or an IP address
+        :param channels: # Channels which generate the waveform
+        :param limits: # min/max V for each channel. if min>max, the polarity of the channel is inverted.
+        """
```

Comment:
You are still missing all the connection and initialization part. It should also check that channels and limits are valid, and report a nice error if not.